pkg/git/git.go

```diff
@@ -263,3 +263,99 @@ func findGitRoot() (string, error) {
 		dir = parent
 	}
 }
+
+func GetFilesystemAt(ref string) (string, error) {
+	repo, err := git.PlainOpenWithOptions(".", &git.PlainOpenOptions{DetectDotGit: true})
+	if err != nil {
+		return "", fmt.Errorf("open git repo: %w", err)
+	}
+
+	h, err := repo.ResolveRevision(plumbing.Revision(ref))
+	if err != nil {
+		return "", fmt.Errorf("resolve ref %q: %w", ref, err)
+	}
+
+	commit, err := repo.CommitObject(*h)
+	if err != nil {
+		return "", fmt.Errorf("read commit: %w", err)
+	}
+
+	tree, err := commit.Tree()
+	if err != nil {
+		return "", fmt.Errorf("read tree: %w", err)
+	}
+
+	// Create temp directory for the git ref
+	tempDir, err := os.MkdirTemp("", "skipctl-git-*")
+	if err != nil {
+		return "", fmt.Errorf("create temp dir: %w", err)
+	}
+
+	if err = writeTreeToDisk(repo, tree, tempDir); err != nil {
+		os.RemoveAll(tempDir)
+		return "", err
+	}
+
+	return tempDir, nil
+}
+
+//nolint:govet,exhaustive,gocognit,mnd // variable shadowing, exhaustive switch, magic numbers acceptable for this helper
+func writeTreeToDisk(repo *git.Repository, tree *object.Tree, targetDir string) error {
```

Comment:
Denne funksjonen fremstår litt spartansk. Har du vurdert å bruke go-git sin clone, fra local path med aktuell ref i stedet for?